src/core/contrib.js

```diff
@@ -71,22 +74,49 @@ export async function run(conf) {
     return;
   }
   const indexes = await response.json();
-  const { issues_url, issue_comment_url, contributors_url } = indexes;
+  const {
+    issues_url,
+    issue_comment_url,
+    comments_url,
+    contributors_url,
+  } = indexes;
+
+  const [
+    issues,
+    issueComments,
+    otherComments,
+    contributors,
+  ] = await Promise.all(
+    [issues_url, issue_comment_url, comments_url, contributors_url].map(url =>
+      fetchAll(
+        new URL(url.replace(/\{[^}]+\}/, ""), window.location.origin).href,
+        headers
+      )
+    )
+  );
 
-  const [issues, comments, contributors] = await Promise.all([
-    fetchIndex(issues_url, headers),
-    fetchIndex(issue_comment_url, headers),
-    fetchIndex(contributors_url, headers),
-  ]);
   const editors = conf.editors.map(nameProp);
-  const commenterUrls = ghCommenters ? findUserURLs(issues, comments) : [];
-  const contributorUrls = ghContributors ? contributors.map(urlProp) : [];
+  const commenterUrls = ghCommenters
+    ? findUserURLs(issues, issueComments, otherComments)
+    : [];
+  const contributorUrls = ghContributors ? contributors.map(URLByUser) : [];
   try {
-    await Promise.all(
-      toHTML(commenterUrls, editors, ghCommenters, headers),
-      toHTML(contributorUrls, editors, ghContributors, headers)
-    );
+    const toHTMLPromises = [];
+    if (ghCommenters) {
```

Comment:
There seems to be a lot of duplication in the following lines... and it looks like it could be `.filter(/* whatever the if is doing */).map( /* to html */)`?

Reply:
Ooh, good idea! I'm a bit concerned that my implementation is overengineered; You be the judge.

Reply:
No, I think it's good. Easy to follow what's going on.